nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java

```diff
@@ -35,14 +35,27 @@
  */
 public interface DocumentBlobManager {
 
+    /**
+     * Reads a {@link Blob} from storage.
+     *
+     * @param blobInfo the blob information
+     * @param doc the document to which this blob belongs
+     * @param xpath the xpath of the blob in the document
+     * @return a managed blob
+     * @since 11.1
+     */
+    Blob readBlob(BlobInfo blobInfo, Document doc, String xpath) throws IOException;
```

Comment:
Maybe a default impl?

Reply:
As it's a service with only one implementation not really worth it, and it has to be implemented anyway.